weights.py

```diff
@@ -0,0 +1,211 @@
+            'MAE': mae,
+            'MSE': mse
+        }
+    else:
+        accuracy, precision, recall, f1 = ([] for _ in range(4))
+        if library == "AutoSklearn":
+            class_names = list(y.unique())
+            class_names.sort()
+            for weight, model in ensemble_model.get_models_with_weights():
+                prediction = model.predict(X)
+                prediction_class = [class_names[idx] for idx in prediction]
+                accuracy.append(round(accuracy_score(y, prediction_class), 2))
+                precision.append(round(precision_score(y, prediction_class, average='micro'), 2))
+                recall.append(round(recall_score(y, prediction_class, average='micro'), 2))
+                f1.append(round(f1_score(y, prediction_class, average='micro'), 2))
+        elif library == "AutoGluon":
+            final_model = ensemble_model.get_model_best()
+            for model_name in ensemble_model.info()['model_info'][ensemble_model.get_model_best()]['stacker_info'][
+                'base_model_names']:
+                ensemble_model.set_model_best(model_name)
+                accuracy.append(round(accuracy_score(y, ensemble_model.predict(X)), 2))
+                precision.append(round(precision_score(y, ensemble_model.predict(X), average='micro'), 2))
+                recall.append(round(recall_score(y, ensemble_model.predict(X), average='micro'), 2))
+                f1.append(round(f1_score(y, ensemble_model.predict(X), average='micro'), 2))
+            ensemble_model.set_model_best(final_model)
+        data = {
+            'weight': weights,
+            'accuracy': accuracy,
+            'precision': precision,
+            'recall': recall,
+            'f1 score': f1
+        }
+
+    return pd.DataFrame(data)
+
+
+def tbl_metrics(ensemble_model, X, y, task, library, weights):
+    df = calculate_metrics(ensemble_model, X, y, task, library, weights)
+    return dash_table.DataTable(
+        data=df.to_dict('records'),  # convert DataFrame to format compatible with dash
+        columns=[
+            {'name': col, 'id': col} for col in df.columns
+        ],
+        style_table={'backgroundColor': '#2c2f38', 'border': '1px solid #2c2f38'},
+        style_cell={
+            'textAlign': 'center',
+            'color': 'white',
+            'border': '1px solid #2c2f38',
+            'backgroundColor': '#1e1e1e',
+            'height': '30px'
+        },
+        id='metrics-table'
+    )
+
+
+def calculate_metrics_adj_ensemble(ensemble_model, X, y, task, library, weights):
+    if task == "regression":
+        mse = round(mean_squared_error(y, ensemble_model.predict(X)))
+        mae = float('%.*g' % (3, mean_absolute_error(y, ensemble_model.predict(X))))
+        mape = float('%.*g' % (3, mean_absolute_percentage_error(y, ensemble_model.predict(X))))
+        predictions = []
+
+        if library == "AutoSklearn":
+            for weight, model in ensemble_model.get_models_with_weights():
```

Comment:
To check if possible: Keeping predictions in the dcc.Store instead of calculating them after every weight modification.